test/distributed/org/apache/cassandra/distributed/impl/Instance.java

```diff
@@ -443,25 +457,31 @@ public void startup(ICluster cluster)
                 StorageService.instance.registerDaemon(CassandraDaemon.getInstanceForTesting());
                 if (config.has(GOSSIP))
                 {
+                    MigrationManager.setUptimeFn(() -> System.currentTimeMillis() - startedAt);
                     StorageService.instance.initServer();
                     StorageService.instance.removeShutdownHook();
                     Gossiper.waitToSettle();
                 }
                 else
                 {
-                    initializeRing(cluster);
-                }
+                    cluster.stream().forEach(peer -> {
+                        // TODO: this isn't going to work for UpgradableCluster
+                        GossipHelper.statusToNormal((IInvokableInstance) peer).accept(this);
+                    });
 
-                StorageService.instance.ensureTraceKeyspace();
+                }
 
                 SystemKeyspace.finishStartup();
 
                 CassandraDaemon.getInstanceForTesting().setupCompleted();
 
                 if (config.has(NATIVE_PROTOCOL))
                 {
+                    // Start up virtual table support
+                    CassandraDaemon.getInstanceForTesting().setupVirtualKeyspaces();
                     CassandraDaemon.getInstanceForTesting().initializeClientTransports();
                     CassandraDaemon.getInstanceForTesting().start();
+                    StorageService.instance.setRpcReady(true);
```

Comment:
can you remove, this doesn't match the behavior of CassandraDaemon, by calling start this will be set IFF native transport is enable, so this should be false if native transport isn't enabled.

Reply:
Done